telemetry.go

```diff
@@ -0,0 +1,16 @@
+package husky
+
+import "context"
+
+// SetAttributesFunc is a function that can be used to set attributes in telemetry controlled
```

Comment:
Hmm naming is hard... maybe something like SetTelemetryAttributes?

Reply:
There's ambiguity: It's not "function that sets attributes", but "assign a function so that when I call SetAttributes it does the thing I want". So perhaps `DefineSetAttributesFunc`?

Or alternatively, rename `SetAttributes` to `AddAttributes` (since it's additive anyway) and call it `SetAddAttributesFunc`?